protocols/luci-proto-openfortivpn/root/usr/libexec/rpcd/luci.openfortivpn

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env lua
+
+local json = require "luci.jsonc"
+local fs   = require "nixio.fs"
+
+local function readfile(path)
+	local s = fs.readfile(path)
+	return s and (s:gsub("^%s+", ""):gsub("%s+$", ""))
+end
+
+local function writefile(path, data)
+	local n = fs.writefile(path, data)
+	return (n == #data)
+end
+
+local function parseInput()
+	local parse = json.new()
+	local done, err
+
+	while true do
+		local chunk = io.read(4096)
+		if not chunk then
+			break
+		elseif not done and not err then
+			done, err = parse:parse(chunk)
+		end
+	end
+
+	if not done then
+		print(json.stringify({ error = err or "Incomplete input" }))
+		os.exit(1)
+	end
+
+	return parse:get()
+end
+
+if arg[1] == "list" then
+	print(json.stringify({
+		getCertificates = {
+			interface = "interface"
+		},
+		setCertificates = {
+			interface = "interface",
+			user_cert = "PEM file data",
+			user_key = "PEM file data",
+			ca_file = "PEM file data"
+		}
+	}))
+elseif arg[1] == "call" then
+	local args = parseInput()
+
+	if not args.interface or
+	   type(args.interface) ~= "string" or
+	   not args.interface:match("^[a-zA-Z0-9_]+$")
+	then
+		print(json.stringify({ error = "Invalid interface name" }))
+		os.exit(1)
+	end
+
+	local user_cert_pem = string.format("/etc/openfortivpn/user-cert-%s.pem", args.interface)
+	local user_key_pem = string.format("/etc/openfortivpn/user-key-%s.pem", args.interface)
+	local ca_file_pem = string.format("/etc/openfortivpn/ca-%s.pem", args.interface)
```

Comment:
Should we not validate if this file does exist?

Reply:
Have added an stat check to `readfile`. Return null if it doesn't exist?